app/jobs/regular/update_username.rb

```diff
@@ -46,9 +46,21 @@ def execute(args)
     def update_posts
       updated_post_ids = Set.new
 
+      # Other people mentioning this user
+      Post
+        .with_deleted
+        .joins(mentioned("posts.id"))
+        .where("a.user_id = :user_id", user_id: @user_id)
+        .find_each do |post|
+          update_post(post)
+          updated_post_ids << post.id
+        end
```

Comment:
This `joins(mentioned())` technique is the approach we used prior to 2e78045af1afa3a3c72dfbf3d072345b26ac83cd. So this is essentially a revert, plus a more surgical fix for the self-mention issue.